ai_traineree/agents/dummy.py

```diff
@@ -0,0 +1,178 @@
+import copy
+import numpy as np
+from ai_traineree.agents import AgentBase
+from ai_traineree.types import AgentState, BufferState, DataSpace, NetworkState
+from typing import Callable, Dict, Optional, Type
+from ai_traineree.loggers import DataLogger
+from ai_traineree.experience import Experience
+
+import warnings
+
+class DummyAgent(AgentBase):
+    """Deep Q-Learning Network (DQN).
+     
+     Agent that returns random values regardless of the input
+     Print mean of the taken actions, observables and reward
+     
+     """
+    model = "Dummy"
+    def __init__(        self,
+        obs_space: DataSpace,
+        action_space: DataSpace,
+        state_transform: Optional[Callable] = None,
+        reward_transform: Optional[Callable] = None,
+        **kwargs
+    ):
+        """Initiates Dummy agent.
+
+        Parameters:
+            obs_space (DataSpace): Dataspace describing the input.
+            action_space (DataSpace): Dataspace describing the output.
+            state_transform (optional func): Function to transform (encode) state before used by the network.
+            reward_transform (optional func): Function to transform reward before use.
+
+        """
+        super().__init__(**kwargs)
+        self.obs_space = obs_space
+        self.action_space = action_space
+        self.iteration: int = 0
+
+        self._loss: float = float("nan")
+
+    @property
+    def loss(self):
+        return {"loss": self._loss}
+
+
+    def reset(self):
+        warnings.warn("Dummy Agent is for debugging purposes only. Nothing to reset.")
+        return
+
+
+    def step(self, exp: None):
+        """Letting the agent to take a step. In this case, since the actions are totally random we don't need to save any observation
+
+        Parameters:
+            obs (ObservationType): Observation.
+            action (int): Discrete action associated with observation.
+            reward (float): Reward obtained for taking action at state.
+            next_obs (ObservationType): Observation in a state where the action took.
+            done: (bool) Whether in terminal (end of episode) state.
+
+        """
+        return
+
+            
+
+
+    def act(self, experience: None, eps: None):
+        """Returns random action.
+
+        Parameters:
+            obs (array_like): current observation
+            eps (optional float): epsilon, for epsilon-greedy action selection. Default 0.
+
+        Returns:
+            Random action taken from the agent.
+
+        """
+        s = 1
+        for i in self.action_space.shape: s*=i
```

Comment:
What is the `s` for? Please be a bit more verbose here. In general I'll discourage from using a single letter variables; they make sense when you write them but not a year later when there's more code intertwine.

Also, break the `for` loop to separate lines. For readability, the code should look similar in the whole code base to ease jumping. Since more often you'll have elaborate `for` than a single line it's better to keep it expanded.

Reply:
Sure thing, I will try to avoid single-letter variables and single line loops. Thanks for the advice 👍